sourcecode/www/src/components/ResourceModal/ResourceStats.jsx

```diff
@@ -19,60 +19,88 @@ const ResourceStats = ({ resourceId }) => {
     const { t } = useTranslation();
     const { edlibApi } = useConfigurationContext();
 
+    const today = moment(new Date()).format('YYYY-MM-DD');
+    const lastMonth = moment(today).subtract(30, 'days').format('YYYY-MM-DD');
+
+    const [startDate, setStartDate] = useState(lastMonth);
+    const [endDate, setEndDate] = useState(today);
+
     const { loading, response } = useFetchWithToken(
-        edlibApi(`/resources/v1/resources/${resourceId}/stats`)
+       edlibApi(`/resources/v1/resources/${resourceId}/stats?start=${startDate}&end=${endDate}`)
     );
 
-    if (loading || !response) {
-        return <CircularProgress />;
-    }
-    const last7daysViews = response.data.last7daysViews;
+    const dateRangeViews = response?.data?.dateRangeViews || {} ;
     const datasets = [
         {
             key: 'count',
             name: _.capitalize(t('resource_view', { count: 2 })),
-            dataset: last7daysViews,
+            dataset: dateRangeViews,
         },
     ];
 
+    const fromDate = moment(startDate);
```

Comment:
`fromDate` and `toDate` are unnecessary and confusing, as they're copies of `startDate` and `endDate`, respectively, but are Moment objects instead of strings. Since Moment can operate on strings, these variables can be removed, and diffDays refactored as:

```js
const diffDays = moment(endDate).diff(fromDate, 'days')
```